Failed to load module (fatal error C1041: cannot open program database '...\vc140.pdb'; if multiple CL.EXE write to the same .PDB file, please use /FS)

The issue only happens after a bug was introduced (tutorial 2) and fixed OR after the execution of the program was paused by a breakpoint.

I disabled Edit and Continue but did not help. Also, I have observed that devenv.exe holds **vc140.pdb**.

/FS flag is set for all the projects for AOT compilation (not sure if that has any effect) and also SetAdditionalCompileOptions("/FS", 0); is called at the beginning of the program.

Environment: Windows 10, VS2017, both Win32 and x64.

I will try to reproduce it on my other computer and report back later.

I can reproduce the issue every time. Here is the log from the VS Output from the point of introducing the BUG in Tutorial 2. Then I hit continue and fixed the bug and saved the file.

```
FileChangeNotifier triggered recompile with changes to:
    File c:\deepui_2017\runtimecompiledcplusplus-master\aurora\examples\simpletest\gameobject.cpp
Compiling...
cl /nologo /Zi /FC /MDd /LDd /Od /FS /D UNICODE /D _UNICODE  /MP /Fo"C:\DeepUI_2017\RuntimeCompiledCPlusPlus-master\Aurora\x64\Debug\Runtime\DEBUG_DEBUG\\" /D WIN32 /EHa /FeC:\Users\Naeron\AppData\Local\Temp\FAD9.tmp  /I "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\runtimeobjectsystem" /I "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\RuntimeCompiler"  "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\examples\simpletest\gameobject.cpp" "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\x64\debug\runtime\debug_debug\objectinterfacepermodulesource.obj" /link /DEBUG:FASTLINK 
echo 
Microsoft Windows [Version 10.0.14393]

(c) 2016 Microsoft Corporation. All rights reserved.

C:\DeepUI_2017\RuntimeCompiledCPlusPlus-master\Aurora\x64\Debug>

**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26403.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86_x64'

gameobject.cpp
   Creating library C:\Users\Naeron\AppData\Local\Temp\FAD9.lib and object C:\Users\Naeron\AppData\Local\Temp\FAD9.exp
The thread 0x3548 has exited with code 0 (0x0).

[RuntimeCompiler] Complete

'SimpleTest.exe' (Win32): Loaded 'C:\Users\Naeron\AppData\Local\Temp\FAD9.tmp'. Symbols loaded.
Compilation Succeeded
Serializing out from 48 old constructors...
Swapping in and creating objects for 1 new constructors...
Serialising in...
Auto Constructing Singletons...
Initialising and testing new serialisation...
Object swap completed
Compile and Module Reload Time: 2.3 s
Have caught an exception in main entity Update loop, code will not be run until new compile - please fix.
FileChangeNotifier triggered recompile with changes to:
    File c:\deepui_2017\runtimecompiledcplusplus-master\aurora\examples\simpletest\gameobject.cpp
Compiling...
cl /nologo /Zi /FC /MDd /LDd /Od /FS /D UNICODE /D _UNICODE  /MP /Fo"C:\DeepUI_2017\RuntimeCompiledCPlusPlus-master\Aurora\x64\Debug\Runtime\DEBUG_DEBUG\\" /D WIN32 /EHa /FeC:\Users\Naeron\AppData\Local\Temp\728B.tmp  /I "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\runtimeobjectsystem" /I "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\RuntimeCompiler"  "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\examples\simpletest\gameobject.cpp" "c:\deepui_2017\runtimecompiledcplusplus-master\aurora\x64\debug\runtime\debug_debug\objectinterfacepermodulesource.obj" /link /DEBUG:FASTLINK 
echo 
Microsoft Windows [Version 10.0.14393]

(c) 2016 Microsoft Corporation. All rights reserved.

C:\DeepUI_2017\RuntimeCompiledCPlusPlus-master\Aurora\x64\Debug>

**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26403.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86_x64'

gameobject.cpp
The thread 0x211c has exited with code 0 (0x0).
c:\deepui_2017\runtimecompiledcplusplus-master\aurora\examples\simpletest\gameobject.cpp: fatal error C1041: cannot open program database 'C:\DeepUI_2017\RuntimeCompiledCPlusPlus-master\Aurora\x64\Debug\vc140.pdb'; if multiple CL.EXE write to the same .PDB file, please use /FS


[RuntimeCompiler] Complete

Failed to load module C:\Users\Naeron\AppData\Local\Temp\728B.tmp

```

--- a/Aurora/RuntimeCompiler/Compiler_PlatformWindows.cpp
+++ b/Aurora/RuntimeCompiler/Compiler_PlatformWindows.cpp
@@ -345,14 +345,6 @@ void Compiler::RunCompile(	const std::vector<FileSystemUtils::Path>&	filesToComp
             linkOptions += " ";
 		}
 	}
-    // faster linking if available: https://randomascii.wordpress.com/2015/07/27/programming-is-puzzles/
-    #if   (_MSC_VER >= 1900)
-        if( linkOptions.empty() )
-        {
-            linkOptions = " /link ";
-        }
-        linkOptions += "/DEBUG:FASTLINK ";
-    #endif
 
 	// Check for intermediate directory, create it if required
 	// There are a lot more checks and robustness that could be added here
@@ -406,10 +398,13 @@ char* pCharTypeFlags = "";
 	pCharTypeFlags = "/D UNICODE /D _UNICODE ";
 #endif
 
+	FileSystemUtils::Path pdbName = moduleName_;
+	pdbName.ReplaceExtension( ".pdb" );
+
 	// /MP - use multiple processes to compile if possible. Only speeds up compile for multiple files and not link
 	std::string cmdToSend = "cl " + flags + pCharTypeFlags
 		+ " /MP /Fo\"" + compilerOptions_.intermediatePath.m_string + "\\\\\" "
-		+ "/D WIN32 /EHa /Fe" + moduleName_.m_string;
+		+ "/D WIN32 /EHa /Fe" + moduleName_.m_string + " /Fd" + pdbName.m_string;
 	cmdToSend += " " + strIncludeFiles + " " + strFilesToCompile + strLinkLibraries + linkOptions
 		+ "\necho ";
 	if( m_pImplData->m_pLogger ) m_pImplData->m_pLogger->LogInfo( "%s", cmdToSend.c_str() ); // use %s to prevent any tokens in compile string being interpreted as formating

